### ISSUE TITLE
blasting to remote database

### ISSUE THREAD

**Issue by jrcunning** [hputnam/Moorea_Sym#12]
I can't seem to get blastn to work using a remote database (for this project or others!). This comes up during the filtering of non-Symbiodinium sequences, which is in the filter_notsym script in the SymITS2 repository. This happens in the current project to generate the "Moorea_sym.RData" phyloseq object from the "Moorea.RData" phyloseq object... however I think ncbi updated their blast tools and server recently and I can no longer get the blasting to work... can you guys try to run "make RAnalysis/Data/Moorea_sym.RData" on your computer to see if you can get it to work? If it says there is nothing to be done, then just run "touch RAnalysis/Data/Moorea.RData" and rerun the make command. On my computer, it begins to run and then the blastn command says "Cannot obtain local host address..." and then sits idle forever...

**Comment by hputnam** [comment]
I get the following error
```BLAST Database error: No alias or index file found for nucleotide database [/Volumes/CoralReefFutures/ref/ncbi_nt/nt] in search path [/Users/hputnam/MyProjects/Moorea_Sym::]```


Also FYI it looks like in the comments there is a hard coding section in the **filter_notSym.R** that  set to a STJ specific file, not to  /Bioinf/clust/poorseqs.fasta

``` # poorseqs <- readDNAStringSet("~/Documents/Academia/HIMB/USVI/STJ2015/data/otus_97_bysample/poorseqs.fasta")```

**Comment by hputnam** [comment]
Is this using an internal file and not the remote?

```# Filter out sequences that are not Symbiodinium from 100% OTUs
RAnalysis/Data/Moorea_sym100.RData: RAnalysis/Data/Moorea_100.RData Bioinf/clust100/100_otus_rep_set.fasta
	R --vanilla < ~/SymITS2/filter_notsym.R --args $^ RAnalysis/Data/Moorea_sym100.RData /Volumes/CoralReefFutures/ref/ncbi_nt/nt```

```# Blast poor matching sequences to local NCBI nt databse, get top hit, write to file
outfile <- file.path(dirname(args[2]), "poorseqs_blast_results.txt")
system(paste("blastn -db", args[4],
            "-query", poorseqsfile, "-num_threads 100",
            "-outfmt '6 qseqid qcovs stitle sseqid' -max_target_seqs 1 | sort -u -k1,1 >",
            outfile))```

**Comment by hputnam** [comment]
I ran the make file using the SymITS2 folder pulled from github today.

**Comment by jrcunning** [comment]
Sorry - I updated the code last night and did not update this thread! I indeed changed the blast database from remote (there were too many issues with the NCBI server not responding, getting overloaded, and/or going really really slow), so I downloaded the entire nt database (50GB) to an external drive and used that. The filter_notsym.R now takes a fourth argument that points to the database to use, so in the Makefile this now points to the nt database on my external drive ("CoralReefFutures"). This is not optimal but is the most functional solution at this point. Unfortunately this means that only I will be able to run this portion of the Makefile. If you want to download the entire nt database somewhere too, then we could each set the same bash environment variable pointing to our copy of the nt database and then use that variable in the Makefile...

Also, yes - the hard path to the STJ2015 file is just a commented line of code I was playing with and haven't deleted from the script yet.

**Comment by hputnam** [comment]
I am downloading to external harddrive now, so I will change the path in the makefile and test.

**Comment by hputnam** [comment]
@jrcunning Not an issue... just thinking out loud RE the pain of the full blast.

How much do you think it matters to the pipeline if the "non-sym" blast is done?  Is it enough to say 
we consider them non-sym at a threshold of X and remove the poorseqs file sequences regardless? I think what you are doing is the most rigorous, but it requires a lot from the inexperienced user if you are thinking about the general utility of your pipeline for the community... but it could also be a good deterrent for those that are tempted to just plug and play...

**Comment by hputnam** [comment]
In defense of keeping the external Blast step... In blasting the poor hits it looks like a mix of coral and Symbiodinium. Could this be a chance to pull more Sym accession numbers into the sym database in an automated fashion?

**Comment by jrcunning** [comment]
I think filtering out the non-sym could be justified simply by a threshold of X... By BLASTing all of these and then potentially *keeping* some of them if their top hit is still *Symbiodinium* then we are allowing more sequences to remain in the dataset. It could be more conservative to just eliminate everything below a certain threshold X (how do we decide X?)... Perhaps by keeping some additional sequences we could add to the database, but I think we'd have to manually QC which ones to add, i.e. depending if they provide a useful name and come from a trusted source...? Also, we may need to add some filtering to the BLAST results from the nt database, I notice that some of them are very low percent identity to their best BLAST hit (incl. some that BLAST to *Symbiodinium*)... I don't currently have a minimum e-value or %id in place on that BLAST search -- do we think one should be added?

**Comment by hputnam** [comment]
https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&amp;id=${acn}&amp;rettype=fasta

**Comment by hputnam** [comment]
I have downloaded and decompressed the nt database for local use. I have changed the path in the make file from CoralReefFutures to my own path.  When I run make I get the following error, which suggests to be is it searching only in the current directory for the nt database and not actually to the path I provided. any thoughts @jrcunning ? Does this have anything to do with the file.path in the filter_notsym.R?

> BLAST Database error: No alias or index file found for nucleotide database [/Volumes/NGS_DATA/ncbi_nt/nt/] in search path [/Users/hputnam/MyProjects/Moorea_Sym::]
